manifests/server/instance/config.pp

```diff
@@ -57,6 +58,7 @@
   $logdir                                          = $postgresql::server::logdir,
   $service_name                                    = $postgresql::server::service_name,
   $service_enable                                  = $postgresql::server::service_enable,
+  Enum[present, absent] $systemd_drop_in_ensure    = $postgresql::server::systemd_drop_in_ensure,
```

Comment:
Does this need to be a separate ensure? I'd say we either have a global instance ensure (absent/present) or no parameter at all.

Reply:
Currently the define does not have an ensure parameter, it is currently just the defined type of the former class. If I add an ensure for the define in the future, I will keep this in mind. Is that a solution which is OK for you @ekohl?

Reply:
Given there is no ensure parameter at all, I'd be tempted to not add this parameter either. Just keep it in the defined type.

I don't see an `@api private` (though perhaps it deserves one?) so I'm always hesitant to add more parameters than needed.

Reply:
@ekohl I totally understand this and thanks for bringing this up. I would like to add an ensure but also want to make sure every resource within the define reflects the status correctly (otherwise its useless). 
And yes, the `@api private` is missing in all the new defines, except the systemd. 
I would like to do both topics in a later PR.

I will remove the `$drop_in_ensure`, except for the systemd defined type.